testing/object_api/test_object_api_c.cpp

```diff
@@ -0,0 +1,230 @@
+ * @brief      Test: parse_args
+ * @details    When given invalid command options,<br>
+ *             parse_args returns FPGA_EXCEPTION.<br>
+ */
+TEST_P(object_api_c_p, parse_args2) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-B");
+  strcpy(two, "3");
+  strcpy(three, "-s");
+
+  char *argv[] = { zero, one, two, three };
+
+  EXPECT_EQ(parse_args(4, argv), FPGA_EXCEPTION);
+}
+
+/**
+ * @test       parse_args3
+ * @brief      Test: parse_args
+ * @details    When given valid command options,<br>
+ *             parse_args populates the global config struct,<br>
+ *             and returns FPGA_OK.<br>
+ */
+TEST_P(object_api_c_p, parse_args3) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-B");
+  strcpy(two, "3");
+
+  char *argv[] = { zero, one, two };
+
+  EXPECT_EQ(parse_args(3, argv), FPGA_OK);
+}
+
+/**
+ * @test       main0
+ * @brief      Test: hello_fpga_main
+ * @details    When given an invalid command option,<br>
+ *             object_api displays an error message,<br>
+ *             and returns non-zero.<br>
+ */
+TEST_P(object_api_c_p, main0) {
+  char zero[20];
+  char one[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-Y");
+
+  char *argv[] = { zero, one };
+
+  EXPECT_NE(object_api_main(2, argv), 0);
+}
+
+INSTANTIATE_TEST_CASE_P(object_api_c, object_api_c_p,
+                        ::testing::ValuesIn(test_platform::platforms({"skx-p"})));
+
+
+class object_api_c_mock_p : public ::testing::TestWithParam<std::string> {
+ protected:
+  object_api_c_mock_p() {}
+
+  virtual void SetUp() override {
+    std::string platform_key = GetParam();
+    ASSERT_TRUE(test_platform::exists(platform_key));
+    platform_ = test_platform::get(platform_key);
+    system_ = test_system::instance();
+    system_->initialize();
+    system_->prepare_syfs(platform_);
+
+    EXPECT_EQ(fpgaInitialize(NULL), FPGA_OK);
+    optind = 0;
+  }
+
+  virtual void TearDown() override {
+    system_->finalize();
+  }
+
+  test_platform platform_;
+  test_system *system_;
+};
+
+/**
+ * @test       main1
+ * @brief      Test: object_api_main
+ * @details    When given a valid command line,<br>
+ *             object_api_main checks *perf directories.<br>
+ *             to add counter, print counters and<br>
+ *             return FPGA_OK.<br>
+ */
+TEST_P(object_api_c_mock_p, main1) {
+  char zero[20];
+  char one[20];
+  char two[20];
+  strcpy(zero, "object_api");
+  strcpy(one, "-B");
+  sprintf(two, "%d", platform_.devices[0].bus);
+
+  char *argv[] = { zero, one, two };
+
+  /*
+  ** FIXME object_api will glob sysfs for *perf directory,
+  ** current test_system does not mock glob and will not
+  ** find *perf/ files,
+  ** but this test should run to successful completion
+  ** on hardware, and the fn should return FPGA_OK.
+  */
+  EXPECT_NE(object_api_main(3, argv), FPGA_OK);
```

Comment:
Shouldn't we duplicate this test case for HW but the assertion would be EXPECT_EQ(main, 0)?